src/explorer/WebAppTreeItem.ts

```diff
@@ -25,13 +27,19 @@ export class WebAppTreeItem extends SiteTreeItem {
     public readonly webJobsNode: IAzureTreeItem;
     public readonly folderNode: IAzureTreeItem;
     public readonly logFolderNode: IAzureTreeItem;
+    public connectionsNode: IAzureTreeItem | undefined;
 
     constructor(client: SiteClient) {
         super(client);
         this.folderNode = new FolderTreeItem(this.client, 'Files', "/site/wwwroot");
         this.logFolderNode = new FolderTreeItem(this.client, 'Log Files', '/LogFiles', 'logFolder');
         this.webJobsNode = new WebJobsTreeItem(this.client);
         this.appSettingsNode = new AppSettingsTreeItem(this.client);
+
+        const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
+        if (workspaceConfig.get(constants.configurationSettings.enableConnectionsNode)) {
```

Comment:
So I wanted you to move _creating_ the node into the constructor, but checking against the workspaceConfig still needs to happen inside of `loadMoreChildren`. The node should always be defined, whether or not it's returned in the array of children below

For example, follow these repro steps:
1. Make sure `enableConnectionsNode` is false
1. Expand a web app
1. Set `enableConnectionsNode` to true
1. Refresh that web app node

The node should show up after the refresh, but I don't think it will with your current code.